h2o-algos/src/main/java/hex/glm/GLM.java

```diff
@@ -170,14 +170,16 @@ public void cv_computeAndSetOptimalParameters(ModelBuilder[] cvModelBuilders) {
         double testDevSq = 0;
         for (int i = 0; i < cvModelBuilders.length; ++i) {  // run cv for each lambda value
           GLM g = (GLM) cvModelBuilders[i];
-          if (g._model._output._submodels[lidx] != null) {
-            double lambda = g._model._output._submodels[lidx].lambda_value;
+          if (g._model._output._submodels.length <= lidx) {
+            double lambda = _parms._lambda[lidx % _parms._lambda.length];
             g._insideCVCheck = true;
             g._driver.computeSubmodel(lidx, lambda, Double.NaN, Double.NaN);
             g._insideCVCheck = false;
-            testDev += g._model._output._submodels[lidx].devianceValid;
-            testDevSq += g._model._output._submodels[lidx].devianceValid * g._model._output._submodels[lidx].devianceValid;
           }
+          assert _parms._lambda[lidx % _parms._lambda.length] == g._model._output._submodels[lidx].lambda_value;
```

Comment:
the submodels with the same `lidx` didn't always have the same lambda value; this was happening for the submodels that were missing in one or more CV models and it seems to be fixed by recomputing only those submodels that were supposed to be recomputed (I added the `assert` to make sure it solves it on more that one problematic dataset).